### PR TITLE
multiplayer is nearly upon us this is the P2P branch

### PR DESCRIPTION
**Pull request by nolram88**
DO NOT MERGE YET PLEASE.

This marks the beginning.

local Vs will operate as normal couch vs.

matchmaking will be the competitive mode. Might integrate with existing services if possible.

P2P will use webRTC to connect players.

server mode will be a hosted machine serving as the point of authority.

### CODE REVIEW

File: src/menus/menu.js [hunk 1]
```diff
@@ -1,25 +1,28 @@
...
   ["VS. Melee", "Target Test", "Target Builder", "Options"],
-  ["Audio", "Gameplay", "Keyboard Controls", "Credits"]
+  ["Audio", "Gameplay", "Keyboard Controls", "Credits"],
+    ["Local VS","Matchmaking","P2P","Server"]
 ];
 const menuExplanation = [
   ["Multiplayer Battles!", "Smash ten targets!", "Build target test stages!", "Game setup."],
-  ["Select audio levels.", "Change gameplay settings.", "Customize keyboard controls.", "Who did this?"]
+  ["Select audio levels.", "Change gameplay settings.", "Customize keyboard controls.", "Who did this?"],
+    ["One box this screen.","Ranked Mode","Hostless Muliplayer","Hosted Multiplayer"]
```
Comment (sheaf) [inline src/menus/menu.js]: "One box this screen."?

Comment (nolram88) [inline src/menus/menu.js]: placeholder i figured someone would question it.

Comment (sheaf) [inline src/menus/menu.js]: :P

File: src/main/main.js [hunk 2]
```diff
@@ -823,6 +828,11 @@ export function gameTick (oldInputBuffers){
   } else if (gameMode == 13) {
     input[creditsPlayer] = interpretInputs(creditsPlayer, true, playerType[creditsPlayer],oldInputBuffers[creditsPlayer]);
     credits(creditsPlayer, input);
+  }else if (gameMode == 14) {
```
Comment (skinnyfoetusboy) [inline src/main/main.js]: [Readability] You should add a space between `}` and `else`

File: src/menus/menu.js (hunk 1 above)
Comment (skinnyfoetusboy) [inline src/menus/menu.js]: [Readability] Indentation is off, please add spaces after commas

File: src/menus/menu.js [hunk 3]
```diff
@@ -1,25 +1,28 @@
...
 ];
-const menuTitle = ["Main Menu", "Options"];
+const menuTitle = ["Main Menu", "Options","Battle Mode"];
```
Comment (skinnyfoetusboy) [inline src/menus/menu.js]: [Readability] Spaces after commas

File: src/menus/menu.js [hunk 4]
```diff
@@ -30,60 +33,119 @@ let menuMode = 0;
...
+          } else {
+            if (menuSelected == SERVERMP) {
+
+            }
+          }
+        }
       }
-    } else {
-      if (menuSelected == 0) {
+    }else {
```
Comment (skinnyfoetusboy) [inline src/menus/menu.js]: [Readability] Space between } and else

**Comment by nolram88** [comment]
looks like we have a flimsy working mp mode

**Comment by nolram88** [comment]
closing this and including it with the other PR

**Pull request closed without merging**